src/common/logs.ts

```diff
@@ -25,7 +26,8 @@ export function createLogGroup(
   },
 ) {
   const shortId = getShortIdOfStack(Stack.of(scope));
-  const logGroupName = `${props.prefix ?? 'clickstream-loggroup'}-${shortId}`;
+  const randStr = generateRandomStr(5, 'abcdefghijklmnopqrstuvwxyz0123456789');
+  const logGroupName = `${props.prefix ?? 'clickstream-loggroup'}-${shortId}${randStr}`;
```

Comment:
What's the use case for the log group name conflict?

Reply:
when manually down-grade analytics stack from latest version to 1.0.x. in AWS console by uploading new cloudformation template, the step function log group name conflict

Reply:
I'm not sure why the replacement resources are happening; is it caused by the changes in resource logical ID?

I don't think it's a good practise to generate a random string in each synthesis time, it will cause replace the log group when updaing the stack.

Reply:
closed this PR.  if it happens again, will do more investigation.